cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -623,8 +631,20 @@ services::Status ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::co
 
     _nClasses = par.nClasses;
 
-    const size_t nRows             = x->getNumberOfRows();
-    const size_t nFeatures         = x->getNumberOfColumns();
+    const size_t nRows     = x->getNumberOfRows();
+    const size_t nFeatures = x->getNumberOfColumns();
+    if (nRows > static_cast<uint32_t>(INT_MAX))
```

Comment:
I suggest to use functionality from service_data_utils.h instead of INT_MAX. Even INT32_MAX looks better relative to INT_MAX in this context,

Reply:
fixed